e3sm_diags/e3sm_diags_driver.py

```diff
@@ -308,11 +370,9 @@ def main(parameters=[]):
         save_provenance(parameters[0].results_dir, parser)
 
     if parameters[0].multiprocessing:
-        parameters = cdp.cdp_run.multiprocess(run_diag, parameters, context="fork")
-    elif parameters[0].distributed:
-        parameters = cdp.cdp_run.distribute(run_diag, parameters)
+        parameters = _run_with_dask_multiprocessing(parameters)
     else:
-        parameters = cdp.cdp_run.serial(run_diag, parameters)
+        parameters = _run_serially(parameters)
 
     parameters = _collapse_results(parameters)
```

Comment:
Replaced `cdp_run` function calls.

I also removed the distributed option because it seemed like it actually never worked and nobody used it.
This run method requires a local Dask cluster setup and a scheduler address (`scheduler_addr` param) passed into `cdp_run.distribute()`.  The `scheduler_addr` is not set anywhere in the codebase for this to work.

More context:
> If that arg wasn't used but this error still appears, then I'm not really sure what's up. If you've noticed, since users only run e3sm_diags either in serial or on a single machine/node with multiprocessing, I don't think dask is even needed. It was only selected since we thought we'd run stuff distributedly.
> - https://github.com/E3SM-Project/e3sm_diags/issues/316#issuecomment-660706642

>distributed: Set to True to run the diagnostics distributedly. It's False by default. multiprocessing and distributed cannot both be set to True. A Dask cluster needs to be up and running. You'll probably never use this.
- https://github.com/E3SM-Project/e3sm_diags/blob/main/docs/source/available-parameters.rst

Reply:
This logic has been refactored in the latest commit.